File: leetcode/String/leetcode-151/151.go
package string

func reverseWords(s string) string {
	if len(s) == 0 {
		return ""
	}

	strs := []string{}

	i := 0
	for i < len(s) {
		if s[i] == ' ' {
			i++
			continue
		} else {
			start := i
			for i < len(s) && s[i] != ' ' {
				i++
			}
			str := s[start:i]
			strs = append(strs, str)
		}
	}

	res := ""
	for i = len(strs) - 1; i >= 0; i-- {
		res += strs[i]
		if i > 0 {
			res += " "
		}
	}
	return res
}

File: leetcode/String/leetcode-151/151.java
class Solution {
    public String reverseWords(String s) {
        if (s == null || s.length() == 0) {
            return "";
        }
        
        String[] strs = s.split(" ");
        StringBuilder res = new StringBuilder();
        for (int i = strs.length - 1; i >= 0; i++) {
            if (!strs[i].equals("")) {
                res.append(strs[i]).append(" ");
            }
        }
        return res.length() == 0 ? "" : res.toString().trim();
    }
}